smartsim/_core/shell/shellLauncher.py

```diff
@@ -0,0 +1,155 @@
+
+
+from __future__ import annotations
+
+import os
+import subprocess as sp
+import typing as t
+
+import psutil
+
+from smartsim._core.dispatch import _EnvironMappingType, _FormatterType, dispatch
+from smartsim._core.utils import helpers
+from smartsim._core.utils.launcher import ExecutableProtocol, create_job_id
+from smartsim.error import errors
+from smartsim.log import get_logger
+from smartsim.settings.arguments.launchArguments import LaunchArguments
+from smartsim.status import JobStatus
+from smartsim.types import LaunchedJobID
+
+if t.TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from smartsim.experiment import Experiment
+
+logger = get_logger(__name__)
+
+
+class ShellLauncher:
+    """Mock launcher for launching/tracking simple shell commands"""
+
+    def __init__(self) -> None:
+        self._launched: dict[LaunchedJobID, sp.Popen[bytes]] = {}
+
+    def start(
+        self, command: tuple[str | os.PathLike[str], t.Sequence[str]]
+    ) -> LaunchedJobID:
+        id_ = create_job_id()
+        path, args = command
+        exe, *rest = args
+        # pylint: disable-next=consider-using-with
+        self._launched[id_] = sp.Popen((helpers.expand_exe_path(exe), *rest), cwd=path)
+        return id_
+
+    def get_status(
+        self, *launched_ids: LaunchedJobID
+    ) -> t.Mapping[LaunchedJobID, JobStatus]:
+        return {id_: self._get_status(id_) for id_ in launched_ids}
+
+    def _get_status(self, id_: LaunchedJobID, /) -> JobStatus:
+        if (proc := self._launched.get(id_)) is None:
+            msg = f"Launcher `{self}` has not launched a job with id `{id_}`"
+            raise errors.LauncherJobNotFound(msg)
+        ret_code = proc.poll()
+        if ret_code is None:
+            status = psutil.Process(proc.pid).status()
+            return {
+                psutil.STATUS_RUNNING: JobStatus.RUNNING,
+                psutil.STATUS_SLEEPING: JobStatus.RUNNING,
+                psutil.STATUS_WAKING: JobStatus.RUNNING,
+                psutil.STATUS_DISK_SLEEP: JobStatus.RUNNING,
+                psutil.STATUS_DEAD: JobStatus.FAILED,
+                psutil.STATUS_TRACING_STOP: JobStatus.PAUSED,
+                psutil.STATUS_WAITING: JobStatus.PAUSED,
+                psutil.STATUS_STOPPED: JobStatus.PAUSED,
+                psutil.STATUS_LOCKED: JobStatus.PAUSED,
+                psutil.STATUS_PARKED: JobStatus.PAUSED,
+                psutil.STATUS_IDLE: JobStatus.PAUSED,
+                psutil.STATUS_ZOMBIE: JobStatus.COMPLETED,
+            }.get(status, JobStatus.UNKNOWN)
+        if ret_code == 0:
+            return JobStatus.COMPLETED
+        return JobStatus.FAILED
+
+    @classmethod
+    def create(cls, _: Experiment) -> Self:
+        return cls()
+
+    @staticmethod
+    def make_shell_format_fn(
+        run_command: str | None,
+    ) -> _FormatterType[
+        LaunchArguments, tuple[str | os.PathLike[str], t.Sequence[str]]
+    ]:
```

Comment:
Not wrong, but I thought I would ask the group for a preference (@juliaputko @amandarichardsonn @mellis13):

Methods like these (that aren't strictly used as helper functions within a class) I tend to prefer to leave as top level functions of a namespace rather than static methods on a class. I think we originally had this as a top level function, but was moved to a `staticmethod` in this PR.

Is there a group/team wide consensus on where methods like these live? Like I mentioned, its mostly just a question of philosophy, but probably best to be consistent!